krystex/src/main/java/com/flipkart/krystal/krystex/node/Node.java

```diff
@@ -86,6 +87,8 @@ class Node {
 
   private final Map<RequestId, Boolean> mainLogicExecuted = new LinkedHashMap<>();
 
+  private final Map<RequestId, Boolean> skipLogicExecuted = new LinkedHashMap<>();
```

Comment:
This should be `skipNodeRequested` which should be populated first thing in the `executeRequestCommand` method